shared/js/dialer/keypad.js

```diff
@@ -836,5 +852,56 @@ var KeypadManager = {
         self._vibrationEnabled = !!value;
       });
     });
+  },
+
+  _updateCaretPosition: function kh_updateCaretPosition() {
+    this._insertPosition = this._realStartPosition();
+  },
+
+  _deleteDigitAtCaret: function kh_deleteDigitAtCaret() {
+    if (this._insertPosition === null) {
+      this._phoneNumber = this._phoneNumber.slice(0, -1);
+    } else {
+      var start = this._realStartPosition();
+      var end = this._realEndPosition();
+      if (start > 0 && start === end) {
+        start = end - 1;
+      }
+      this._phoneNumber = this._phoneNumber.substring(0, start) +
+                          this._phoneNumber.substring(end);
+      this._insertPosition = start;
+    }
+  },
+
+  restoreCaretPosition: function kh_restoreCaretPosition() {
+    if (this._insertPosition !== null) {
+      var caretPosition = this._caretPosition(this._insertPosition);
+      this.phoneNumberView.selectionStart = caretPosition;
+      this.phoneNumberView.selectionEnd = caretPosition;
+      this.phoneNumberView.focus();
+    }
+  },
+
```

Comment:
Oh, I see this comment in the comment view and I did not see you were referring to the **following** functions. Sorry.